searxng/searxng
# Fix: autocomplete with the POST method: url encode the user query

## What does this PR do?

See the commit and the comment in the code: currently, the endpoint `/autocompleter` receives `1 1` when the users type `1+1`. This is a bug in autocomplete.js.

Since the project seems [abandoned](https://github.com/autocompletejs/autocomplete.js/issues/78), the PR monkey patch a function of the project. 

## Why is this change important?





## How to test this PR locally?

With the POST, and then with the GET method: check the autocompletion of `1 + 1 !`.
You should see in both cases `1 + 1 !images` and not `1 1 !images` as the first suggestion.

## Author's checklist

https://www.npmjs.com/package/@tarekraafat/autocomplete.js might be a good replacement for autocomplete.js

## Related issues

Partially related to https://github.com/searxng/searxng/issues/1695
